_rules/focus-returns-trigger-9au0ou.md

```diff
@@ -0,0 +1,144 @@
+---
+id: 9au0ou
+name: Focus returns to trigger
+rule_type: atomic
+description: |
+  This rule checks that when a modal closes the focus returns to the trigger that opened the modal.
+accessibility_requirements:
+  wcag20:2.4.3: # Focus Order (A)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+  wcag-technique:G59: # Placing the interactive elements in an order that follows sequences and relationships within the content
+    forConformance: false
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+input_aspects:
+  - Accessibility Tree
+  - DOM Tree
+acknowledgments:
+  authors:
+    -  Helen Burge
+  previous_authors:
+---
+
+## Applicability
+
+This rule applies to any [modal][] that when closed the [focus][] returns to the [trigger][].
+
+## Expectation
+
+On closing the [modal][], the assistive technology [focus][] will return to the item in the owning document that triggered the [modal][].
```

Comment:
I see two difficulties here:
- we need to tie the "open" button/link/… with the modal, which we can't really do if we target the modal (there is little way to know how it was opened);
- this sounds like a process rather than a static check, that's OK but that is also something we do not have in rules so far (they check a static snapshot of the page).

---
My suggestion would be to have an Applicability on the trigger, which with our definition could be "an [instrument](https://act-rules.github.io/glossary/#instrument-to-achieve-an-objective) to open a modal", and an Expectation in the line of "after opening + closing, the focus is back on the trigger that was used for opening".
(that would also catch cases where several buttons can open the same modal but focus always go back to the same, rather than the one that was used)

We still need to be careful about how we phrase the process, since it is something we do not have currently in our rules…

I think we can also define "modal" as "role of `dialog` + `aria-modal="true"`" to get a less ambiguous definition 🤔 And "open" would be something about making it visible.

Reply:
All good points, and yes - I need to work on it more 👍 

I like your suggestions. I will work on this in the next few days.